packages/webview_flutter/lib/src/tizen_webview.dart

```diff
@@ -0,0 +1,228 @@
+    _invokeChannelMethod<void>('loadRequest', <String?, String?>{'url': uri});
+  }
+
+  /// Accessor to the current URL that the WebView is displaying.
+  Future<String?> currentUrl() => _invokeChannelMethod<String>('currentUrl');
+
+  /// Checks whether there's a back history item.
+  Future<bool> canGoBack() =>
+      _invokeChannelMethod<bool>('canGoBack').then((bool? result) => result!);
+
+  /// Checks whether there's a forward history item.
+  Future<bool> canGoForward() => _invokeChannelMethod<bool>('canGoForward')
+      .then((bool? result) => result!);
+
+  /// Goes back in the history of this WebView.
+  Future<void> goBack() => _invokeChannelMethod<void>('goBack');
+
+  /// Goes forward in the history of this WebView.
+  Future<void> goForward() => _invokeChannelMethod<void>('goForward');
+
+  /// Reloads the current URL.
+  Future<void> reload() async {
+    _invokeChannelMethod<void>('reload');
+  }
+
+  /// Clears all caches used by the [WebView].
+  Future<void> clearCache() async {
+    _invokeChannelMethod<void>('clearCache');
+  }
+
+  /// Sets the JavaScript execution mode to be used by the webview.
+  Future<void> setJavaScriptMode(int javaScriptMode) async {
+    _invokeChannelMethod<void>('javaScriptMode', javaScriptMode);
+  }
+
+  /// Returns the title of the currently loaded page.
+  Future<String?> getTitle() {
+    return _invokeChannelMethod<String>('getTitle');
+  }
+
+  /// Set the scrolled position of this view.
+  Future<void> scrollTo(int x, int y) async {
+    _invokeChannelMethod<void>('scrollTo', <String, int>{
+      'x': x,
+      'y': y,
+    });
+  }
+
+  /// Move the scrolled position of this view.
+  Future<void> scrollBy(int x, int y) {
+    return _invokeChannelMethod<void>('scrollBy', <String, int>{
+      'x': x,
+      'y': y,
+    });
+  }
+
+  /// Return the set scroll position of this view.
+  Future<Offset> getScrollPosition() async {
+    final dynamic position =
+        await _invokeChannelMethod<dynamic>('getScrollPosition');
+    if (position == null) {
+      return Offset.zero;
+    }
+    return Offset(position['x'] as double, position['y'] as double);
+  }
+
+  /// Set the current background color of this view.
+  Future<void> setBackgroundColor(Color color) async {
+    _invokeChannelMethod<void>('backgroundColor', color.value);
+  }
+
+  /// Set whether controller have a NavigationDelegate.
+  Future<void> setHasNavigationDelegate(bool hasNavigationDelegate) async {
+    _hasNavigationDelegate = hasNavigationDelegate;
+  }
+
+  /// Adds a new JavaScript channel to the set of enabled channels.
+  Future<void> addJavaScriptChannel(
+      JavaScriptChannelParams javaScriptChannelParams) async {
+    // When JavaScript channel with the same name exists make sure to remove it
+    // before registering the new channel.
+    if (_javaScriptChannelParams.containsKey(javaScriptChannelParams.name)) {
+      _javaScriptChannelParams.remove(javaScriptChannelParams.name);
+    }
```

Comment:
This code effectively does nothing.

c.f. The Android implementation supports removing existing JS channels.

Reply:
removed

Reply:
We're not ever going to support removing already added JS channels, right?

Reply:
Yes, there is no api supporting JS channel remove api in ewk. That code is my mistake. It's right to delete it as you said.